pymeasure/instruments/lecroy/lecroyT3DSO1204.py

```diff
@@ -0,0 +1,1151 @@
+        with _ChunkResizer(self.adapter, 20 * 1024 * 1024):
+            img = self.binary_values("SCDP", dtype=np.uint8)
+        return bytearray(img)
+
+    def download_data(self, source, requested_points=0, sparsing=1, averaging=1):
+        """ Get data points from the specified source of the oscilloscope. Ther returned objects are
+        two np.ndarray of data and time points and a dict with the waveform preamble, that contains
+        metadata about the waveform.
+        Note.
+        :param source: measurement source. It can be "C1", "C2", "C3", "C4", "MATH.
+        :param requested_points: number of points to acquire. If 0 or None all available points
+        will be returned.
+        :param sparsing: interval between data points. For example if sparsing = 4, only one
+        point every 4 points is read.
+        :param averaging: average over data points. For example if averaging = 4, only the 4-point
+        average value is returned.
+        :return: data_ndarray, time_ndarray, waveform_preamble_dict: see waveform_preamble
+        property for dict format. """
+
+        def _header_sanity_checks(message):
+            # The format of the header is DAT2,#9XXXXXXX where XXXXXXX is the number of acquired
+            # points and it is zero padded.
+            message_header = bytes(message[0:self._header_size]).decode("ascii")
+            # Sanity check on header and footer
+            if message_header[0:7] != "DAT2,#9":
+                raise ValueError(f"Waveform data in invalid : header is {message_header}")
+            return int(message_header[-9:])
+
+        def _npoints_sanity_checks(message):
+            message_header = bytes(message[0:self._header_size]).decode("ascii")
+            transmitted_points = int(message_header[-9:])
+            received_points = len(message) - self._header_size - self._footer_size
+            if transmitted_points != received_points:
+                raise ValueError(f"Number of transmitted points ({transmitted_points}) != "
+                                 f"number of received points ({received_points})")
+
+        def _footer_sanity_checks(message):
+            # The footer is always a double line-carriage \n\n
+            message_footer = bytes(message[-self._footer_size:]).decode("ascii")
+            if message_footer != "\n\n":
+                raise ValueError(f"Waveform data in invalid : footer is {message_footer}")
+
+        # First of all set the correct acquisition type
+        if averaging > 1:
+            self.acquisition_type = "average"
+            self.acquisition_average = averaging
+        else:
+            self.acquisition_type = "normal"
+
+        # Check that we are trying to read a reasonable amount of points
+        sample_points = self.acquisition_sample_size(self.waveform_source)
+        if requested_points > 0 and 0 < sample_points < requested_points:
```

Comment:
that condition will raise a Value error, if someone sets `requested_points=None` as you offer in the docstring, as you cannot compare None.
The most simple solution is to remove None from the docstring

Reply:
Thank you for spotting that inconsistency. Fix applied in commit 21c9431cdb444c39e9aa0db8ea3a740dd5f620f3

Reply:
resolving